RabbitMQ.Stream.Client/StreamSystem.cs

```diff
@@ -342,10 +343,12 @@ public async Task<StreamInfo> StreamInfo(string streamName)
                     case DnsEndPoint { Host: "localhost" } dnsEndPoint:
                         forceLocalHost = true;
                         localPort = dnsEndPoint.Port;
+                        localHostOrAddress = dnsEndPoint.Host;
                         break;
-                    case IPEndPoint ipEndPoint when Equals(ipEndPoint.Address, IPAddress.Loopback):
+                    case IPEndPoint ipEndPoint when IPAddress.IsLoopback(ipEndPoint.Address):
```

Comment:
`IPAddress.IsLoopback` will support both IPV4 and IPV6

>In the case of IPv4, that the [IsLoopback](https://learn.microsoft.com/en-us/dotnet/api/system.net.ipaddress.isloopback?view=net-8.0) method returns true for any IP address of the form 127.X.Y.Z (where X, Y, and Z are in the range 0-255), not just [Loopback](https://learn.microsoft.com/en-us/dotnet/api/system.net.ipaddress.loopback?view=net-8.0#system-net-ipaddress-loopback) (127.0.0.1).

https://learn.microsoft.com/en-us/dotnet/api/system.net.ipaddress.isloopback?view=net-8.0